ProcessLib/HeatTransportBHE/LocalAssemblers/HeatTransportBHELocalAssemblerSoil-impl.h

```diff
@@ -114,16 +122,72 @@ void HeatTransportBHELocalAssemblerSoil<ShapeFunction, IntegrationMethod>::
                     MaterialPropertyLib::PropertyType::specific_heat_capacity)
                 .template value<double>(vars, pos, t);
 
-        auto const density_s =
-            solid_phase.property(MaterialPropertyLib::PropertyType::density)
+        auto const density_f =
+            liquid_phase.property(MaterialPropertyLib::PropertyType::density)
+                .template value<double>(vars, pos, t);
+
+        auto const heat_capacity_f =
+            liquid_phase
+                .property(
+                    MaterialPropertyLib::PropertyType::specific_heat_capacity)
+                .template value<double>(vars, pos, t);
+
+        auto const porosity =
+            medium.property(MaterialPropertyLib::PropertyType::porosity)
                 .template value<double>(vars, pos, t);
-        // for now only using the solid phase parameters
+
+        auto const velocity_arr =
+            liquid_phase
+                .property(MaterialPropertyLib::PropertyType::phase_velocity)
+                .value(vars, pos, t);
+        auto const velocity = Eigen::Map<Eigen::Vector3d const>(
+            std::get<MaterialPropertyLib::Vector>(velocity_arr).data(), 1, 3);
+
+        // calculate the hydrodynamic thermodispersion tensor
+        auto const thermal_conductivity =
+            MaterialPropertyLib::formEigenTensor<3>(
+                medium
+                    .property(
+                        MaterialPropertyLib::PropertyType::thermal_conductivity)
+                    .value(vars, pos, t));
+
+        auto thermal_conductivity_dispersivity = thermal_conductivity;
+
+        double const velocity_magnitude = velocity.norm();
+
+        if (velocity_magnitude >= std::numeric_limits<double>::epsilon())
+        {
+            auto const thermal_dispersivity_longitudinal =
+                medium
+                    .property(MaterialPropertyLib::PropertyType::
+                                  thermal_longitudinal_dispersivity)
+                    .template value<double>();
+            auto const thermal_dispersivity_transversal =
+                medium
+                    .property(MaterialPropertyLib::PropertyType::
+                                  thermal_transversal_dispersivity)
+                    .template value<double>();
+
+            auto const thermal_dispersivity =
+                density_f * heat_capacity_f *
+                (thermal_dispersivity_transversal * velocity_magnitude * I +
```

Comment:
minor; I'd use Eigen::Identity directly here, or even directly construct a diagonal matrix with the value *e.g.*
```c++
Eigen::Vector3d::Constant(Dt * v).asDiagonal() + ...
```

Reply:
Replaced with Eigen::Matrix3d::Identity()